### ISSUE TITLE
Upgrade Plans

### ISSUE THREAD

**Issue by nasirkhan** [nasirkhan/laravel-starter#269]
There are a number of places where improvements are needed and more features can be added too. I a listing a few here and there are more. We may discuss those here plan and prepare a work plan as well.

- [ ] Image Gallery Feature for Articles Module. (Create a new Image Gallery and add those in any part of the Posts)
- [ ] Improve Documentation (Create docs for all the features)
- [ ] Page Builder (Find or build a visual page builder, to make the pages easily)
- [x] Improved Module Manager (I started to create a new package and will publish it soon.)
- [ ] Theme Manager (Find an easier way to use new themes)
- [x] Add More SEO features across the project, especially for the Articles Module
- [x] Remove the Profile view and update feature from User Module
- [x] Update `datetime picker` // https://getdatepicker.com
- [x] Improve frontend theme
- [ ] Multisite: https://github.com/gecche/laravel-multidomain could be an option (Ref. https://github.com/nasirkhan/laravel-starter/issues/380#issuecomment-1404123176)
- [x] Visual Menu Manager. 

> Everyone is welcome to contribute to this project. You can Work on the current project, and new features listed here and propose something new as well. Let's discuss here and improve this Laravel Starter.

**Comment by MusheAbdulHakim** [comment]
which of these will you like me to work on.

**Comment by mohamadkarammoussa** [comment]
hello 
1- dont forget add to your plan export excel pdf and print 
2- add multi lang for content
 ... i hope to you  good job

**Comment by nasirkhan** [comment]
Thanks for you comments. 

This is a general purpose starter project and I assume Excel/ Spreadsheet or PDF export might necessary for all the projects. There are two very good laravel package available for exporting data and content in multiple formats. 

Multilingual content option would be nice to have. But it requires some planning on which approach should be taken to implement this feature. There are a number of ways to implement his, can you suggest me which one should we use here?

**Comment by mohamadkarammoussa** [comment]
hello 
thanks for your comments.
i used https://github.com/spatie/laravel-translatable in one of my projects .. it was good

**Comment by stormtr** [comment]
This project can add Tenant would be perfect for multi web sites

**Comment by nasirkhan** [comment]
> This project can add Tenant would be perfect for multi web sites

Thank you for your suggestion. Multi tenancy is not required for most of the applications. That is why I did not planned to add this. And again, multi tenancy can be implemented in a number of ways, so pushing towards in one direction might not be a good idea.

**Comment by chirumist** [comment]
hi there is one other issue 
when we use react, vue, how to use module base components, if anyone want to use inertia? how to manage this?

**Comment by chirumist** [comment]
This project doesn't have multiple module scripts & SCSS build. Is there any plan?

**Comment by chirumist** [comment]
I have implemented this dashboard. Also, I added their setting customization on the setting page.

https://templates.iqonic.design/hope-ui/html/dist/dashboard/index.html

**Comment by nasirkhan** [comment]
@chirumist Thank you your comments. I believe you found this project useful for your own projects. 

The goal for this project is to make it as simple as possible. React, Vue, Interia are awesome ways to build applications, but it is not for those who just started Laravel. So I consciously avoided that path. There are some livewire components and I was thinking to add more in future.  

Multiple build for modules can be implemented, but that will add complicity. The core modules do not need much changes. For the additional custom modules any build setup can be used. 

The dashboard is beautiful. Is it open source? 

If you are extending this laravel starter, I would suggest you to send pull requests to improve this project. As you implemented the dashboard, you may work on a feature to easily change the theme both in Backend and Frontend.

**Comment by chirumist** [comment]
Yes, this dashboard is open source. When I got some time then I will contribute to this project.

**Comment by ghost** [comment]
Hello Nasir!

I was just checking the project and found that the feature you mentioned Theme Manager is not yet implemented. There are number of options available. I was checking that as well. But I thought of giving you my feedback:

When using multiple themes, it would be utmost important to have domain/site mapping. If not, it becomes not very useable. So I think having a manual entry of domain or subdomain name would be extremely helpful or necessary.

I checked one multisite package based on env replicattion. This is the most simple and easiest option. I was too surprised with that simplicity. One can simply configure multisite in few minutes with a seperate env generator.

This seems to me the best combinitation because the multisite does not touch any other areas and provides the simplest option. It only needs to be integrated with theme manager.

One these both packages are integrated, this template is going to be much better. The multilanguage option is now there. But the multisite is not yet implemented, though we can do ourselves.

To have sites like en.site.com, de.site.com, fr.site.com can help. Here the theme manager could give a greate help, if you could integrate.

**Comment by boedyirh** [comment]
The Laravel Starter is very helpfull.
I was wondering if We can extend the list page, with more advance filter. (individual filer or active filter).


https://filamentexamples.com/storage/62/jcH1YQMlg3Pzwa7k9Su4K7CpBUCwyW-metaMTgtdGFibGUtd2l0aC1jb21wbGV4LWZpbHRlcnMucG5n-.png

**Comment by nasirkhan** [comment]
@boedyirh Thank you for your suggestion. It would be really helpful to have this feature. But unfortunately, I might not be able to work on this feature soon. If you have already developed this feature, please submit PR. 

> The Laravel Starter is very helpfull. I was wondering if We can extend the list page, with more advance filter. (individual filer or active filter).
> 
> https://filamentexamples.com/storage/62/jcH1YQMlg3Pzwa7k9Su4K7CpBUCwyW-metaMTgtdGFibGUtd2l0aC1jb21wbGV4LWZpbHRlcnMucG5n-.png

**Comment by boedyirh** [comment]
Thanks, I will surely do that
